engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/writers/TypedStreamWriterImpl.java

```diff
@@ -129,4 +129,29 @@ public void configureSourceContext(final long sourceRecordPosition) {
   public void appendFollowUpEvent(final long key, final Intent intent, final RecordValue value) {
     appendRecord(key, RecordType.EVENT, intent, value);
   }
+
+  /**
+   * Use this to know whether you can add an event of the given length to the underlying batch
+   * writer.
+   *
+   * @param eventLength the length of the event that will be added to the batch
+   * @return true if an event of length {@code eventLength} can be added to this batch such that it
+   *     can later be written
+   */
+  @Override
+  public boolean canWriteEventOfLength(final int eventLength) {
+    return batchWriter.canWriteAdditionalEvent(eventLength);
+  }
+
+  /**
+   * This is not actually accurate, as the frame length needs to also be aligned by the same amount
+   * of bytes as the batch. However, this would break concerns here, i.e. the writer here would have
+   * to become Dispatcher aware.
```

Comment:
:thought_balloon: I guess the comment breaks the abstraction boundary as well :clown_face: